src/nft/components/collection/TransactionCompleteModal.tsx

```diff
@@ -0,0 +1,363 @@
+  const shouldShowModal = (txState === TxStateType.Success || txState === TxStateType.Failed) && txState
+  const {
+    nftsPurchased,
+    nftsNotPurchased,
+    showPurchasedModal,
+    showRefundModal,
+    totalPurchaseValue,
+    totalRefundValue,
+    totalUSDRefund,
+    txFeeFiat,
+  } = useMemo(() => {
+    let nftsPurchased: UpdatedGenieAsset[] = []
+    let nftsNotPurchased: UpdatedGenieAsset[] = []
+    let showPurchasedModal = false
+    let showRefundModal = false
+    let totalPurchaseValue = BigNumber.from(0)
+    let totalRefundValue = BigNumber.from(0)
+    let totalUSDRefund = 0
+    let txFeeFiat = 0
+
+    if (transactionResponse !== undefined) {
+      const { nftsPurchased: purchasedNfts, nftsNotPurchased: notPurchasedNfts, txReceipt } = transactionResponse
+      if (nftsPurchased && nftsNotPurchased && txReceipt) {
+        nftsPurchased = purchasedNfts
+        nftsNotPurchased = notPurchasedNfts
+        showPurchasedModal = nftsPurchased.length >= 1
+        showRefundModal = nftsNotPurchased.length >= 1
+        totalPurchaseValue = getTotalNftValue(nftsPurchased)
+        totalRefundValue = getTotalNftValue(nftsNotPurchased)
+        totalUSDRefund = totalRefundValue && parseFloat(formatEther(totalRefundValue)) * ethPrice
+        const txFee = BigNumber.from(txReceipt ? txReceipt.gasUsed : 0).mul(
+          BigNumber.from(txReceipt ? txReceipt.effectiveGasPrice : 0)
+        )
+        txFeeFiat = parseFloat(formatEther(txFee)) * ethPrice
+      }
+    }
+
+    return {
+      nftsPurchased,
+      nftsNotPurchased,
+      showPurchasedModal,
+      showRefundModal,
+      totalPurchaseValue,
+      totalRefundValue,
+      totalUSDRefund,
+      txFeeFiat,
+    }
+  }, [transactionResponse, ethPrice])
+
+  const toggleShowUnavailable = () => {
+    setShowUnavailable(!showUnavailable)
+  }
+
+  function closeTxCompleteScreen() {
+    setTransactionResponse({} as TxResponse)
+    setTxState(TxStateType.New)
+  }
+
+  // Given the length of the array of successfully purchased NFTs, returns the maxHeight and maxWidth of each asset preview
+  const getSuccessfulImageSize = (numSuccessful: number) => {
+    const sizeModifier = isMobile ? 2 : 1
+    if (numSuccessful === 1) {
+      return 574 / sizeModifier
+    } else if (numSuccessful === 2) {
+      return 280 / sizeModifier
+    } else if (numSuccessful === 3 || (numSuccessful >= 5 && numSuccessful < 7)) {
+      return 184 / sizeModifier
+    } else if (numSuccessful === 4 || (numSuccessful >= 7 && numSuccessful < 13)) {
+      return 136 / sizeModifier
+    } else if (numSuccessful >= 13 && numSuccessful < 21) {
+      return 108 / sizeModifier
+    } else return isMobile ? 39 : 64
+  }
+
+  useEffect(() => {
+    fetchPrice().then((price) => {
+      setEthPrice(price ?? 0)
+    })
+  }, [])
+
+  useEffect(() => {
+    useSendTransaction.subscribe((state) => (transactionStateRef.current = state.state))
+  }, [])
+
+  return (
```

Comment:
non-blocking but we could make this file significantly more readable if we split the modal components below into 'SuccessModal', 'MixedRefundModal', and 'FullRefundModal'